core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreUtil.java

```diff
@@ -111,4 +118,114 @@ public static Certificate getCertificate(String alias, KeyStore store) throws Ax
+                    RegistryResources.SecurityManagement.CustomKeyStore.KEYSTORE_PREFIX.length());
+        }
+
+        OMElement config = null;
+        try {
+            OMElement customKeyStoresConfigElem = XMLUtils.toOM(serverConfiguration.getDocumentElement())
+                    .getFirstChildWithName(KeyStoreUtil.getQNameWithCarbonNS(
+                            RegistryResources.SecurityManagement.CustomKeyStore.ELEM_SECURITY))
+                    .getFirstChildWithName(KeyStoreUtil.getQNameWithCarbonNS(
+                            RegistryResources.SecurityManagement.CustomKeyStore.ELEM_CUSTOM_KEYSTORES));
+
+            Iterator<OMElement> iterator = customKeyStoresConfigElem.getChildElements();
+            while (iterator.hasNext()) {
+                OMElement customKeyStoreConfig = iterator.next();
+                String location = customKeyStoreConfig.getFirstChildWithName(KeyStoreUtil.getQNameWithCarbonNS(
+                        RegistryResources.SecurityManagement.CustomKeyStore.PROP_LOCATION)).getText();
+
+                if (location.endsWith(keyStoreName)) {
+                    config = customKeyStoreConfig;
+                    break;
+                }
+            }
+        } catch (Exception e) {
+            throw new CarbonException("Error occurred while reading custom keystore configuration.", e);
+        }
+
+        if (config == null) {
+            throw new CarbonException("No configuration found for custom key store : " + keyStoreName);
+        }
+
+        return config;
+    }
+
+    /**
+     * This method returns the requested key store property value from the configuration.
+     *
+     * @param config                OMElement object of configuration.
+     * @param propertyName                  Property name.
+     * @return Configuration value as String.
+     * @throws CarbonException      If failed to retrieve the requested configuration.
+     */
+    public static String getCustomKeyStoreConfig(OMElement config, String propertyName) throws CarbonException {
+
+        validateKeyStoreConfigName(propertyName);
+
+        String configValue = config.getFirstChildWithName(getQNameWithCarbonNS(propertyName)).getText();
+
+        if (RegistryResources.SecurityManagement.CustomKeyStore.PROP_LOCATION.equals(propertyName)) {
+            // Replace carbon.home placeholder with proper location path
+            if (configValue.startsWith(RegistryResources.SecurityManagement.CARBON_HOME_PLACEHOLDER)) {
+                configValue = configValue.replace(RegistryResources.SecurityManagement.CARBON_HOME_PLACEHOLDER, "");
+                configValue = new File(".").getAbsolutePath() + configValue;
+            } else {
+                throw new CarbonException("Invalid key store location: " + configValue);
+            }
+        }
+
+        return configValue;
+    }
+
+    public static void validateKeyStoreConfigName(String propertyName) throws CarbonException {
```

Comment:
add the java doc comments

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/3aa02c8fd1024563e7aefe9329049af4802dfac3